### PR TITLE
Backend channel numbers matrix

### PR DESCRIPTION
**Pull request by phunkyfish**
v4.3.0
- Added: Support backend channel numbers for all channel groups not just the first
- Added: Ignore empty channel groups
- Added: Readme updates
- Fixed: Revert support hidden entries for backend channel numbers
- Fixed: Fix hanging on deleting multiple recordings at once

### CODE REVIEW

File: pvr.vuplus/resources/language/resource.language.en_gb/strings.po [hunk 1]
```diff
@@ -937,12 +937,12 @@ msgstr ""
 
 #help: General - updateint
 msgctxt "#30625"
-msgid "As the set-top box can also be used to modify timers, delete recordings etc. and the set-top box does not notify the Kodi installation, the addon needs to regularly check for updates (new channels, new/changed/deletes timers, deleted recordings, etc.) This property defines how often the addon checks for updates."
+msgid "As the set-top box can also be used to modify timers, delete recordings etc. and the set-top box does not notify the Kodi installation, the addon needs to regularly check for updates (new channels, new/changed/deletes timers, deleted recordings, etc.) This property defines how often the addon checks for updates. Please note that updating the recordings frequently can keep your receiver and it's harddisk from entering standby automatically."
```
Comment (ksooo) [inline pvr.vuplus/resources/language/resource.language.en_gb/strings.po:940]: Change seems unrelated to the commit message "Readme updates ".

File: pvr.vuplus/resources/language/resource.language.en_gb/strings.po [hunk 2]
```diff
@@ -937,12 +937,12 @@ msgstr ""
...
 msgstr ""
 
 #help: General - updatemode
 msgctxt "#30626"
-msgid "The mode used when the update interval is reached. Note that if there is any timer change detected a recordings update will always occur regardless of the update mode. Choose from one of the following two modes: [Timers and Recordings] Update all timers and recordings; [Timers only] Only update the timers."
+msgid "The mode used when the update interval is reached. Note that if there is any timer change detected a recordings update will always occur regardless of the update mode. Choose from one of the following two modes: [Timers and Recordings] Update all timers and recordings; [Timers only] Only update the timers. If it's important to not spin up the HDD on your STB use this option. The HDD should then only spin up when a timer event occurs."
```
Comment (ksooo) [inline pvr.vuplus/resources/language/resource.language.en_gb/strings.po:945]: Change seems unrelated to the commit message "Readme updates ".

File: src/enigma2/ChannelGroups.h [hunk 3]
```diff
@@ -49,6 +49,7 @@ namespace enigma2
     void ClearChannelGroups();
     std::vector<std::shared_ptr<enigma2::data::ChannelGroup>>& GetChannelGroupsList();
     bool LoadChannelGroups();
+    std::vector<std::shared_ptr<enigma2::data::ChannelGroup>>& GetExtraDataChannelGroupsList() { return m_extraDataChannelGroups; };
```
Comment (ksooo) [inline src/enigma2/ChannelGroups.h]: Can this method be const?

File: src/enigma2/Channels.cpp [hunk 4]
```diff
@@ -148,6 +148,17 @@ bool Channels::LoadChannels(ChannelGroups &channelGroups)
       bOk = true;
   }
 
+  // Load Channels extra data for groups
+  int tvChannelNumberOffset = 0;
+  int radioChannelNumberOffset = 0;
+  for (auto& group : channelGroups.GetExtraDataChannelGroupsList())
```
Comment (ksooo) [inline src/enigma2/Channels.cpp]: `const auto& group`?

File: pvr.vuplus/resources/language/resource.language.en_gb/strings.po (hunk 1 above)
Comment (phunkyfish) [reply pvr.vuplus/resources/language/resource.language.en_gb/strings.po:940]: Renamed commit

File: pvr.vuplus/resources/language/resource.language.en_gb/strings.po (hunk 2 above)
Comment (phunkyfish) [reply pvr.vuplus/resources/language/resource.language.en_gb/strings.po:945]: Renamed commit

File: src/enigma2/ChannelGroups.h (hunk 3 above)
Comment (phunkyfish) [reply src/enigma2/ChannelGroups.h]: No, as it returns a vector.

File: src/enigma2/Channels.cpp (hunk 4 above)
Comment (phunkyfish) [reply src/enigma2/Channels.cpp]: Done

File: src/enigma2/ChannelGroups.h (hunk 3 above)
Comment (ksooo) [reply src/enigma2/ChannelGroups.h]: Huh? Cannot see how this should be related.

Comment (ksooo) [reply src/enigma2/ChannelGroups.h]: The returned vector reference could even be const, not only the method, if  I read the code correctly.

Comment (phunkyfish) [reply src/enigma2/ChannelGroups.h]: You're right, I didn't make the returned reference const at the same time!

Comment (phunkyfish) [reply src/enigma2/ChannelGroups.h]: Ooops

**Comment by phunkyfish** [comment]
@ksooo it’s ready now. The Leia PR mirrors this.

**Comment by phunkyfish** [comment]
@ksooo added one additional minor fix. Should be good now.

File: src/enigma2/Recordings.cpp [hunk 5]
```diff
@@ -418,7 +418,8 @@ PVR_ERROR Recordings::DeleteRecording(const PVR_RECORDING &recinfo)
   if(!WebUtils::SendSimpleCommand(strTmp, strResult))
     return PVR_ERROR_FAILED;
 
-  PVR->TriggerRecordingUpdate();
+  // No need to call PVR->TriggerRecordingUpdate() as it is handled by kodi PVR.
+  // In fact when multiple recordings are remvoed at once, calling it here can cause hanging issues
```
Comment (ksooo) [inline src/enigma2/Recordings.cpp]: Typo: remvoed ;-)

Are there stack traces for the "hanging" issue? I would like to investigate that.

Comment (phunkyfish) [reply src/enigma2/Recordings.cpp]: Fixed

**Comment by ksooo** [comment]
Not sure this should better be fixed in Kodi core. Although it might be not nessacery to call `TriggerRecordingsUpdate` here, Kodi definitely should not hang if this is done nevertheless.

**Comment by phunkyfish** [comment]
Happy to revert after @bservet can test it as a potential fix. As his platform doesn’t get updated often with new addons versions there should be time to pull the required stacktraces / crash logs. But if you would prefer this be removed anyway just ask.

**Comment by ksooo** [comment]
> But if you would prefer this be removed anyway just ask.

 No need to remove, imo.

**Comment by phunkyfish** [comment]
Cool, should be ready to go (again)!

**Comment by phunkyfish** [comment]
@ksooo approval ping 😉

**Review by ksooo — APPROVED** [review]
